lib/mayaUsd/render/vp2RenderDelegate/mesh.cpp

```diff
@@ -895,6 +899,14 @@ void HdVP2Mesh::Sync(
             _meshSharedData->_allRequiredPrimvars.push_back(HdTokens->points);
 
         _UpdatePrimvarSources(delegate, *dirtyBits, _meshSharedData->_allRequiredPrimvars);
+
+        // update the type of vertex layout to use (shared/unshared)
+        bool requireUnsharedVertexLayout
+            = _IsUnsharedVertexLayoutRequired(_meshSharedData->_primvarInfo);
+        if (_meshSharedData->_isVertexLayoutUnshared != requireUnsharedVertexLayout) {
+            _meshSharedData->_isVertexLayoutUnshared = requireUnsharedVertexLayout;
+            _ResetRenderingTopology();
```

Comment:
A change in primvars may change our choice of vertex layout (shared/unshared) and then we would need to regenerate the rendering topology